infoblox/resource_infoblox_ip_allocation.go

```diff
@@ -68,6 +69,30 @@ func resourceIPAllocation() *schema.Resource {
 				Required:    true,
 				Description: "Unique identifier of your tenant in cloud.",
 			},
+			"extattr": &schema.Schema{
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "Map of extensible attributes.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"values": {
+							Type:     schema.TypeList,
```

Comment:
Map allows as of now only TypeBool, TypeInt, TypeFloat and TypeString as possible value data types [https://github.com/hashicorp/terraform/blob/6697245c18dad6848e7647598dda1ab04c2680ca/internal/legacy/helper/schema/schema.go#L1605](url) and [https://github.com/hashicorp/terraform/blob/v0.14.7/helper/schema/schema.go#L169](url)

I would expect the plugin would crash at run time though it does not give a compilation error.

To overcome this you can use jsonencode at tf file and then unmarshal it at code end [https://github.com/infobloxopen/terraform-provider-infoblox/blob/develop/infoblox/resource_infoblox_network.go#L93](url)

Reply:
@somashekhar  i don't understand your point 

> I would expect the plugin would crash at run time though it does not give a compilation error.

This is why we have acceptance tests, and it's passing it.

Regarding 
> Map allows as of now only TypeBool, TypeInt, TypeFloat and TypeString as possible value data types

This is *Resource not TypeMap

>  If it is *Resource, the element type is a complex structure,
https://github.com/hashicorp/terraform/blob/fce896d6df314a2c77a9daecb5c61d8beea157db/helper/schema/schema.go#L173

Regarding 
> To overcome this you can use jsonencode at tf file and then unmarshal it at code end

in my opinion, I don't think setting the values as json content in tf is a recommended way, the users should have a simple way to define the parameters values.

the way you're suggesting will do marshal (tf code using jsonencode) , then unmarshal (resource), then marshal again (client side) to send the request using rest api